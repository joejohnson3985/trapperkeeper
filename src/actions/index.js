export const setCards = (cards) => ({
  type: 'SET_CARDS',
  notes
})

export const createCard = (card) => ({
  type: 'CREATE_CARD',
  card
})