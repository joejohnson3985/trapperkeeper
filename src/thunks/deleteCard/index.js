import { removeCard } from '../../actions';
import fetchData from '../fetchData';

export const deleteCard = (id) => {
  return async (dispatch) =>  {
    try {
      const url = `http://localhost:3000/api/v1/cards/${id}`;
      const options = {
        method: 'DELETE'
      }
      await fetchData(url, options);
      dispatch(removeCard(id));
    } catch (error) {

    }
  }
}

export default deleteCard;