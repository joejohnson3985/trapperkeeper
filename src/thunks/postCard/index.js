import { createCard } from '../../actions';
import { fetchData } from '../fetchData';

export const postCard = (state) => {
  return async (dispatch) =>  {
    try {
      const url = 'http://localhost:3000/api/v1/cards';
      const options = {
        method: 'POST',
        body: JSON.stringify(state),
        headers: {
          'Content-Type': 'application/json'
        }
      }
      const newCard = await fetchData(url, options);
      dispatch(createCard(newCard));
    } catch (error) {

    }
  }
}

export default postCard;