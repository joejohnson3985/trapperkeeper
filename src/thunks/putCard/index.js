import { updateCard } from '../../actions';
import fetchData from '../fetchData';

export const putCard = (card) => {
  return async (dispatch) =>  {
    try {
      const url = `http://localhost:3000/api/v1/cards/${card.id}`;
      const options = {
        method: 'PUT',
        headers: {
          'Content-Type': 'application/JSON'
        },
        body: JSON.stringify(card)
      }
      await fetchData(url, options);
      dispatch(updateCard(card));
    } catch (error) {

    }
  }
}

export default putCard;