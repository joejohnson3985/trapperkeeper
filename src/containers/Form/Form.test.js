import React from 'react';
import { Form, mapStateToProps, mapDispatchToProps } from './index';
import * as actions from '../../actions/index';
import { shallow, mount } from 'enzyme';

import putCard from '../../thunks/putCard/index';
jest.mock('../../thunks/putCard')

import deleteCard from '../../thunks/deleteCard';
jest.mock('../../thunks/deleteCard')

import postCard from '../../thunks/postCard';
jest.mock('../../thunks/postCard')


describe('Form', () => {

  describe('Component', () => {
    let wrapper;
    let mockData;

    beforeEach(() => {

      mockData = {id:1, name:'chores', list: [1,2,3]}

      wrapper = shallow(
        <Form cardData={mockData}
              id={mockData.id}
              list={mockData.list} />
      )
    });

    it('should have a snapshot', () => {
      console.log(wrapper)
      expect(wrapper).toMatchSnapshot()
    });

    it('should have default state', () => {
      wrapper = shallow(<Form />)
      expect(wrapper.state()).toEqual({
        name:'',
        item:'',
        list:[]
      })
    });

  });
  describe('mapDispatchToProps', () => {
    it('should call method postCard with correct params', () => {
      let mockCard = {id:1, name:'chores', list:[1,2,3]}
      let mockDispatch = jest.fn();
      let actionToDispatch = postCard(mockCard)
      let mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.postCard(actionToDispatch)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('should call method putCard with correct params', () => {
      let mockCard = {id:1, name:'chores', list:[1,2,3]}
      let mockDispatch = jest.fn();
      let actionToDispatch = putCard(mockCard)
      let mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.putCard(actionToDispatch)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('should call method deleteCard with correct params', () => {
      let mockId = 12323123
      let mockDispatch = jest.fn();
      let actionToDispatch = deleteCard(mockId)
      let mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.deleteCard(actionToDispatch)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  });
});


