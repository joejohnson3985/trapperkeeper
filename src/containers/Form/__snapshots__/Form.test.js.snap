// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should have a snapshot 1`] = `
<div
  className="overlay"
>
  <form
    className="Form"
  >
    <input
      id="new-title-input"
      name="name"
      onChange={[Function]}
      placeholder="Add Title"
      type="text"
      value="Grocery List"
    />
    <FormItem
      checked={false}
      handleItemSubmit={[Function]}
      item="Coke"
      key="321312"
      list_id={321312}
      removeItem={[Function]}
    />
    <input
      className="item-text"
      contentEditable={true}
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="List item"
      type="text"
      value=""
    />
    <FormItem
      checked={true}
      handleItemSubmit={[Function]}
      item="Pop"
      key="123123"
      list_id={123123}
      removeItem={[Function]}
    />
    <FormItem
      checked={true}
      handleItemSubmit={[Function]}
      item="Soda"
      key="231312"
      list_id={231312}
      removeItem={[Function]}
    />
    <Link
      className="home-button"
      onClick={[Function]}
      to="/"
    >
      Save
    </Link>
    <Link
      className="home-button"
      to="/"
    >
      Home
    </Link>
    <Link
      onClick={[Function]}
      to="/"
    >
      <img
        alt="Delete card"
        src="delete_outline.svg"
      />
    </Link>
  </form>
</div>
`;
